src/main/java/athena/TimeAllocator.java

```diff
@@ -53,60 +58,153 @@ public void runAllocate() throws TaskNotFoundException {
-                while (temp.contains(-1)) {
-                    for (Task currTask : undefinedTimeTasks) {
-                        Integer span = currTask.getTimeInfo().getDuration();
-                        if (span <= space) {
-                            int taskNumber = currTask.getNumber();
-                            this.flexibleTaskList.getTaskFromNumber(taskNumber)
-                                    .getTimeInfo().setStartTime(LocalTime.of(end - space, 0));
-                            for (int i = 0; i < span; i++) {
-                                temp.set(i + end - space - pos, currTask.getNumber());
-                            }
-                            space -= span;
-                            tasksToRemove.add(currTask);
-                            if (space == 0) {
-                                break;
-                            }
-                        }
-                    }
-                    removeAssignedTasks(undefinedTimeTasks, tasksToRemove);
-                    setTempToPerm(dayLog, temp);
-                    if (start == end) {
-                        break;
-                    } else {
-                        start = end;
-                    }
-                    if (undefinedTimeTasks.size() == 0) {
-                        done = true;
-                        break;
-                    }
+                undefinedTimeTasks = carryOverTasks;
+                start = end;
+                if (end == sleep) {
+                    done = true;
+                }
+            }
+        }
+    }
+
+    /**
+     * Assigns the best time to the Tasks.
+     *
+     * @param bestLog list of taskNumbers in the index corresponding to the hour they are assigned to
+     * @param pos     starting position of the log
+     * @throws TaskNotFoundException if task is not in the taskList
+     */
+    private void assignTime(ArrayList<Integer> bestLog, int pos) throws TaskNotFoundException {
+        int count = 0;
+        ArrayList<Integer> assignedNumbers = new ArrayList<>();
+        for (int taskNumber : bestLog) {
+            if (!assignedNumbers.contains(taskNumber)) {
+                this.flexibleTaskList.getTaskFromNumber(taskNumber)
+                        .getTimeInfo().setStartTime(LocalTime.of(pos + count, 0));
+                assignedNumbers.add(taskNumber);
+                System.out.println(taskNumber);
+                System.out.println(pos + count);
+            }
+            count++;
+        }
+    }
+
+    /**
+     * Creates the arrangement that reduces the breaks in the timetable.
+     * Iterates through the undefinedTimeTasks
+     * Modifies the excess tasks left over
+     *
+     * @param pos                staring position
+     * @param end                ending position
+     * @param undefinedTimeTasks Tasks with undefined times
+     * @param carryOverTasks     Tasks that did not get assigned yet
```

Comment:
I think it would be better to say that this method will store the tasks that did not get assigned yet into this parameter. Otherwise, it seems like the caller has to provide said tasks.